WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationProtobufMessageProcessor.swift

```diff
@@ -0,0 +1,174 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPI
+import WireDataModel
+import WireProtos
+
+// sourcery: AutoMockable
+/// A common processor for processing MLS / Proteus protobuf message.
+/// Used by `ConversationMLSMessageAddEventProcessor` and `ConversationProteusMessageAddEventProcessor`
+/// The message content is encoded using protocol buffers. There is a common protocol buffer definition adopted by all Wire client.
+public protocol ConversationProtobufMessageProcessorProtocol {
+
+    func processProtobufMessage(
+        _ message: GenericMessage,
+        content: GenericMessage.OneOf_Content,
+        conversation: ZMConversation,
+        conversationID: ConversationID,
+        senderID: UserID,
+        senderClientID: String?,
+        logAttributes: LogAttributes,
+        date: Date
+    ) async
+
+}
+
+struct ConversationProtobufMessageProcessor: ConversationProtobufMessageProcessorProtocol {
+
+    let messageLocalStore: any MessageLocalStoreProtocol
+    let userLocalStore: any UserLocalStoreProtocol
+
+    func processProtobufMessage(
+        _ message: GenericMessage,
+        content: GenericMessage.OneOf_Content,
+        conversation: ZMConversation,
+        conversationID: ConversationID,
+        senderID: UserID,
+        senderClientID: String?,
+        logAttributes: LogAttributes,
+        date: Date
+    ) async {
+        var logAttributes = logAttributes
+        WireLogger.eventProcessing.debug("Processing:\n\(message)")
+        logAttributes[.nonce] = UUID(uuidString: message.messageID) ?? "<nil>"
+        WireLogger.eventProcessing.debug("Processing message", attributes: logAttributes)
+
+        // Message content types: https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/20545866/Messages
+        switch content {
+        case .lastRead:
+
+            await messageLocalStore.updateLastReadMessageTimestamp(
+                message.lastRead,
+                in: conversation
+            )
```

Comment:
This modifies the conversation, so I would expect to see this method in the conversation local store:

```swift
// Something like...
await conversationLocalStore.updateLastReadTimestamp(
  conversationID: message.lastRead.conversationID,
  lastReadTimestamp: message.lastRead.timestamp
)
```

Reply:
yes you're right, fixed in https://github.com/wireapp/wire-ios/pull/2164/commits/90da7addc7e950bb179e65ed21fe6583b88d9f4c